redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -25,133 +25,144 @@ import app.cash.redwood.Modifier
 import platform.Foundation.classForCoder
-import platform.UIKit.UICollectionView
-import platform.UIKit.UICollectionViewCell
-import platform.UIKit.UICollectionViewDataSourceProtocol
-import platform.UIKit.UICollectionViewDelegateFlowLayoutProtocol
-import platform.UIKit.UICollectionViewFlowLayout
-import platform.UIKit.UICollectionViewFlowLayoutAutomaticSize
-import platform.UIKit.UICollectionViewLayout
-import platform.UIKit.UICollectionViewLayoutAttributes
-import platform.UIKit.UICollectionViewScrollDirection.UICollectionViewScrollDirectionHorizontal
-import platform.UIKit.UICollectionViewScrollDirection.UICollectionViewScrollDirectionVertical
-import platform.UIKit.UICollectionViewScrollPositionTop
 import platform.UIKit.UIColor
 import platform.UIKit.UIControlEventValueChanged
 import platform.UIKit.UIEdgeInsetsMake
 import platform.UIKit.UIRefreshControl
 import platform.UIKit.UIScrollView
+import platform.UIKit.UITableView
+import platform.UIKit.UITableViewAutomaticDimension
+import platform.UIKit.UITableViewCell
+import platform.UIKit.UITableViewCellSeparatorStyle.UITableViewCellSeparatorStyleNone
+import platform.UIKit.UITableViewCellStyle
+import platform.UIKit.UITableViewDataSourceProtocol
+import platform.UIKit.UITableViewDelegateProtocol
+import platform.UIKit.UITableViewRowAnimationNone
+import platform.UIKit.UITableViewScrollPosition
 import platform.UIKit.UIView
 import platform.UIKit.indexPathForItem
 import platform.UIKit.item
 import platform.darwin.NSInteger
 import platform.darwin.NSObject
 
 internal open class UIViewLazyList(
-  private var collectionViewFlowLayout: UICollectionViewFlowLayout = UICollectionViewFlowLayout().apply {
-    estimatedItemSize = UICollectionViewFlowLayoutAutomaticSize.readValue()
-  },
-  internal val collectionView: UICollectionView = UICollectionView(
+  internal val tableView: UITableView = UITableView(
     CGRectZero.readValue(),
-    collectionViewFlowLayout,
   ),
-) : WindowedLazyList<UIView>(UICollectionViewListUpdateCallback(collectionView)), ChangeListener {
+) : LazyList<UIView>, ChangeListener {
+  override var modifier: Modifier = Modifier
 
-  // Fetch the item relative to the entire collection view
-  private fun WindowedChildren<UIView>.itemForGlobalIndex(index: Int): Widget<UIView> {
-    return get(index) ?: placeholder[index % placeholder.size]
-  }
+  override val value: UIView
+    get() = tableView
+
+  protected var onViewportChanged: ((firstVisibleItemIndex: Int, lastVisibleItemIndex: Int) -> Unit)? = null
+
+  private val processor = object : LazyListUpdateProcessor<LazyListContainerCell, UIView>() {
+    override fun createCell(
+      cell: Cell<LazyListContainerCell, UIView>,
+      widget: Widget<UIView>,
+      index: Int,
+    ): LazyListContainerCell {
+      val result = tableView.dequeueReusableCellWithIdentifier(
+        identifier = reuseIdentifier,
+        forIndexPath = NSIndexPath.indexPathForItem(index.convert(), 0.convert()),
```

Comment:
Calling `indexPathForItem` this frequently might be expensive--it might be worth trying to just create our own index paths, and enforcing that the table view only has a single section (so that every index path is initialized with `.init(section: 0, row: index)`